srplasticity/srp_extrafuncs.py

```diff
@@ -390,91 +498,140 @@ def plot_srp(model, target_dict, stimulus_dict, protocols=None):
             fig.panels[ix].plot(xax, mean, color="#cc3311", label="SRP model")
+            fig.panels[ix].fill_between(xax, lower_SD, upper_SD, color="xkcd:light grey", label="SD")
             if protocols != None:
                 fig.panels[ix].set_title(protocols[key])
             else:
                 fig.panels[ix].set_title(key)
-            fig.panels[ix].set_xticks(xax)
+            if len(xax) <= 10:
+                fig.panels[ix].set_xticks(xax)
+            else:
+                ticks = np.arange(1, len(mean) + 1, math.ceil(len(mean) / 10))
+                fig.panels[ix].set_xticks(ticks)
             fig.panels[ix].set_ylim(0.5, 9)
-            fig.panels[ix].set_yticks([1, 3, 5, 7, 9])
+            fig.panels[ix].set_yticks([1, 3, 5, 7, 9, 11])
 
         fig.panels[0].legend(frameon=False)
         fig.panels[0].set_ylabel("norm. EPSC amplitude")
 
         plt.show()
 
 
-def plot_estimates(means):
+def plot_estimates(means, efficacies):
+    """
+    Plot sample estimates over time with the corresponding mean values.
+
+    :param means: A list of mean predicted responses
+    :type means: list
+    :param efficacies: A list of sample predicted responses
+    :type efficacies: list
+    """
+
     fig, axis = plt.subplots()
-    xax = range(len(means))
-    axis.plot(range(1, len(xax) + 1), means, lw=1, color="#cc3311")
+
+    if type(efficacies[0]) != np.float64 and type(efficacies[0]) != float:
+        xax = range(len(efficacies[0]))
+
+        for i in range(len(efficacies)):
+            if i == 0:
+                axis.plot(range(1, len(xax) + 1), efficacies[i], lw=1, color="#cc3311", label="Samples")
+            else:
+                axis.plot(range(1, len(xax) + 1), efficacies[i], lw=1, color="#cc3311")
+
+        axis.plot(range(1, len(xax) + 1), means, lw=1.2, color="black", label='Means')
+
+    else:
+        xax = range(len(efficacies))
+        axis.plot(range(1, len(xax) + 1), efficacies, lw=1, color="#cc3311", label="Samples")
+        axis.plot(range(1, len(xax) + 1), means, lw=1.2, color="black", label='Means')
+
     axis.spines['top'].set_visible(False)
     axis.spines['right'].set_visible(False)
     axis.set_ylabel("Predicted EPSC", labelpad=1, size=8)
     axis.set_xticks(range(1, len(xax) + 1))
     axis.set_xlabel("spike nr.", labelpad=1)
-    fig.set_dpi(1200)
-    fig.tight_layout()
-    # plt.savefig(f"estimates_plot.svg", transparent=True)
+    fig.legend(frameon=False)
 
 
 def plot_spike_train(spiketrain):
+    """
+    Plot a spike train
+
+    :param spiketrain: A binary stimulation vector from a
+                        vector with ISI intervals to be plotted
+    :type spiketrain: Numpy array
+    """
+
     fig, axis = plt.subplots()
     axis.plot(spiketrain, lw=0.7, color='black')
-    axis.set_ylim(-0.005, 0.005)
+    axis.set_ylim(-1e-5, 5e-6)
     axis.axis("off")
 
-    fig.set_dpi(1200)
-    fig.tight_layout()
-    # plt.savefig(f"spike_train_plot.svg", transparent=True)
```

Comment:
This function does not return, save, or show anything.

Reply:
I added plt.show() to all the plotting functions, the users can decide outside of them if they want to save the plot